File: demo/src/main/java/com/example/demo/service/ProductService.java
package com.example.demo.service;

import com.example.demo.dto.ProductDTO;
import com.example.demo.dto.response.ProductResponse;
import com.example.demo.entity.Product;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Set;


public interface ProductService {
    public ProductDTO addProduct(ProductDTO productDTO);
    public ProductDTO updateProduct(Long id, ProductDTO productDTO);
    public Product getProductById(Long id);
    public List<ProductDTO> getAllProducts();
    public boolean deleteProduct(Long id);
    public Page<ProductDTO> getAllProductsPaged(int page, int size, String sortBy, String direction);
    public ProductResponse getById(Long id);
    Page<ProductDTO> searchProductsByName(String name, int page, int size);
    Page<ProductDTO> filterProductsBySellPrice(double minPrice, double maxPrice, int page, int size, String sortBy, String direction);
    Page<ProductDTO> filterProductsByCategoryNative(Set<Long> categoryIds, int page, int size, String sortBy, String direction);
    Page<ProductDTO> filterProductsByCategoryAndPriceNative(Set<Long> categoryIds, double minPrice, double maxPrice, int page, int size, String sortBy, String direction);
    List<Product> add(List<Product> products);
    public Page<ProductResponse> getAll(Double sellPrice, Integer status, Integer page, Integer size);
    public Page<ProductResponse> getAllByCategorySlug(String slug,Double sellPrice, Integer status, Integer page, Integer size);

}

File: document/backup/camerashop_fe1/services/ProductService.ts
import axios from 'axios';
import { ProductDTO } from '../types';

const API_URL = '/api/products';

export const getProducts = async (): Promise<ProductDTO[]> => {
    const response = await axios.get(API_URL);
    return response.data;
};

export const getProductById = async (id: number): Promise<ProductDTO> => {
    const response = await axios.get(`${API_URL}/${id}`);
    return response.data;
};

export const createProduct = async (product: ProductDTO): Promise<ProductDTO> => {
    const response = await axios.post(API_URL, product);
    return response.data;
};

export const updateProduct = async (id: number, product: ProductDTO): Promise<ProductDTO> => {
    const response = await axios.put(`${API_URL}/${id}`, product);
    return response.data;
};

export const deleteProduct = async (id: number): Promise<void> => {
    await axios.delete(`${API_URL}/${id}`);
};
